src/components/utils/PositionInScreen/hooks.ts

```diff
@@ -45,6 +52,10 @@ const useHeights = (
     const childResizeObserver = new ResizeObserver((entries) => {
       const dropdown = head(entries);
 
+      if (dropdown?.target === dropdown?.contentRect) {
```

Comment:
this function and :25 are very similar, I would suggest changing them to one to maintain one code with a callback action for your sideEffect

```
resizeObserverHandler = (callback) => new ResizeObserver((entries) => {
      const parent = head(entries);

      /** If the event is triggered by the element itself then return
       * to avoid an infinite loop
       */
      if (parent?.target === parent?.contentRect) {
        return;
      }

      if (parent) {
        callback(parent.contentRect.height);
      }
    });
```

and use of it

```
const childResizeObserver = resizeObserverHandler(setChildHeight);
...

```